### DISCUSSION TITLE
How to add timeout back to ui.notification after long calculations

### DISCUSSION THREAD

**Discussion by zilch42** [zauberzeug/nicegui#2379] [Q&A] [answered]
### Question

I'd like to use `ui.notification` and keep it up while a long calculation is running, then dismiss it after a few seconds once it's done (so the user can actually see the done message). 

I can keep the notification indefinitely by setting `timeout=None`. I'd have thought I should be able to update the `timeout` later in the same way that the `message`, `spinner` and `icon` can be updated, but it doesn't seem to work. In the code below, the "done" message stays up indefinitely.

```python
import asyncio
from nicegui import ui


async def compute():
    n = ui.notification(message="Computing...", spinner=True, type="ongoing", timeout=None)
    await asyncio.sleep(10)
    n.message = "Done"
    n.spinner = False
    n.icon = 'done'
    n.timeout = 5.0

ui.button('Compute', on_click=compute)

ui.run()
```

**Comment by rodja** [comment, marked as answer]
Oh, dismissing a notification was not possible. I created pull request https://github.com/zauberzeug/nicegui/pull/2380 to change that in the next release. Then you can do

```py
async def compute():
    n = ui.notification(message="Computing...", spinner=True, type="ongoing", timeout=None)
    await asyncio.sleep(1)
    n.message = "Done"
    n.spinner = False
    n.icon = 'done'
    await asyncio.sleep(1)
    n.dismiss()

ui.button('Compute', on_click=compute)
```

**Reply by zilch42** [reply]
Awesome thanks!

**Reply by rodja** [reply]
We just released 1.4.11 with this feature 🚀
